### PR TITLE
replace easybuild templates with f-strings on Java easyconfigs

### LINKED ISSUE #22090: Failure due to templating in checksums of extension
Test are failing in #22088 for `Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb` due to the usage of a `%(version)s` template in the checksum specification of an extension:

```
== 2024-12-21 00:31:11,868 easyblock.py:871 INFO Found file Sys-Info-Driver-Linux-0.7905.tar.gz at /project/def-maintainers/boegelbot/sources/p/Perl-bundle-CPAN/extensions/Sys-Info-Driver-Linux-0.7905.tar.gz
== 2024-12-21 00:31:11,868 filetools.py:1265 DEBUG Using blocksize 16777216 for calculating the checksum
== 2024-12-21 00:31:11,869 easyblock.py:692 INFO sha256 checksum for /project/def-maintainers/boegelbot/sources/p/Perl-bundle-CPAN/extensions/Sys-Info-Driver-Linux-0.7905.tar.gz: 899c329bd3508ec5849ad0e5dadfa7c3679bbacaea9dda12404a7893032e8b7b
== 2024-12-21 00:31:11,869 easyblock.py:695 DEBUG Verifying checksums for extension source...
== 2024-12-21 00:31:12,039 build_log.py:226 ERROR EasyBuild encountered an error (at easybuild/5.0.x/easybuild-framework/easybuild/base/exceptions.py:126 in __init__): Missing checksum for Sys-Info-Driver-Linux-0.7905.tar.gz in {'Sys-Info-Driver-Linux-%(version)s.tar.gz': '899c329bd3508ec5849ad0e5dadfa7c3679bbacaea9dda12404a7893032e8b7b'} (at easybuild/5.0.x/easybuild-framework/easybuild/tools/filetools.py:1307 in verify_checksum)
```

This is the easyconfig code related to this error:
https://github.com/easybuilders/easybuild-easyconfigs/blob/ef644ae1e1edeab766f0c9632bdd8a210186737e/easybuild/easyconfigs/p/Perl-bundle-CPAN/Perl-bundle-CPAN-5.36.1-GCCcore-12.3.0.eb#L2120-L2135

### PR DESCRIPTION
**Pull request by lexming**
Fixes #22090 for Java easyconfigs

### CODE REVIEW

File: easybuild/easyconfigs/j/Java/Java-11.0.16.eb [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-
 # Using the Adoptium Eclipse Temurin builds, recommended by https://whichjdk.com/#distributions
-
-source_urls = ['https://github.com/adoptium/temurin%%(version_major)s-binaries/releases/download/jdk-%%(version)s+%s/'
-               % local_build]
-sources = [local_tarball_tmpl % ('%(jdkarch)s', local_build)]
-
+_tarball_tmpl = f'OpenJDK{_java_version}U-jdk_{{}}_linux_hotspot_{version}_{_build_version}.tar.gz'
+sources = [_tarball_tmpl.format('%(jdkarch)s')]
+source_urls = [
```
Comment (boegel) [inline easybuild/easyconfigs/j/Java/Java-11.0.16.eb:16]: @lexming We usually have `source_urls` above `sources`, why need keep doing that?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/j/Java/Java-11.0.16.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/j/Java/Java-11.0.16.eb]: yeah I know, but scripting these changes late at night is hard :slightly_smiling_face:

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen3
EB_BRANCH=5.0.x
EB_ARGS="Java-8.402.eb Java-11.0.20.eb Java-17.0.6.eb Java-19.0.2.eb Java-21.0.5.eb"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 14 out of 14 (14 easyconfigs in total)
node3507.doduo.os - Linux RHEL 8.8, x86_64, AMD EPYC 7552 48-Core Processor, Python 3.6.8
See https://gist.github.com/boegel/7044a92a6597e5f618e7493866b75d6c for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**